CMakeLists.txt

```diff
@@ -255,6 +255,7 @@ endif()
 # Link test executable against gtest
 add_executable(IntegrationTestsRunner
     "source/tests/utilities/run_all_tests.cpp"
+    "source/tests/integration/ni_fake_service_tests_endtoend.cpp"
```

Comment:
Should we consider a `source/tests/endtoend` or `source/tests/acceptance` instead of appending `endtoend` to the test files? And possibly a new test runner? Or will this be the only `endtoend` test source?

Reply:
Good question. I think the `integration` test category that these are currently in is a good one. In this context, end-to-end is a subcategory of Integration tests that run an in-process server and connect a client to it. Unlike system tests (which are all end-to-end) they don't have external deps. There might be a more-perfect name for that category. I think your suggestions are less differentiated from 'system', which I don't like.

These files technically don't need an end-to-end suffix. There's another file with the same name, but it's in a different directory. I actually think that's helpful though. Both because it's easier for navigation and because it quickly (if partially) answers the question "what is a fake integration test?"

For fake tests, we only use end-to-end tests when our primary unit-testing strategy doesn't work. This is pretty rare. If we decided to make this the primary fake testing strategy, maybe like a FakeEndToEnd runner would be a better option.